compositor_render/src/transformations/layout/apply_layouts.wgsl

```diff
@@ -0,0 +1,62 @@
+struct VertexInput {
+    @location(0) position: vec3<f32>,
+    @location(1) tex_coords: vec2<f32>,
+    @location(2) layout_id: i32,
+}
+
+struct VertexOutput {
+    @builtin(position) position: vec4<f32>,
+    @location(0) tex_coords: vec2<f32>,
+    @location(1) @interpolate(flat) layout_id: i32,
+}
+
+
+struct Layout {
+    position_transformation: mat4x4<f32>,
+    texture_id: i32,
+    background_color: vec4<f32>,
+}
```

Comment:
Some comments explain that `background_color` is only used when `texture_id==-1` would be helpful.
Also `background_color` is probably not the most suitable name here. Without looking at the implementation, I expect that this shader receives n textures with backgrounds and renders those backgrounds behind if texture size doesn't match/fill the whole position of a single layout/rectangle.

`position_transformation` is also misleading a bit, since this not only changes the position of the box, but also its rotation. I would use `box_transformation`.

Since `wgsl` doesn't support union types, ideally I would define [const](https://www.w3.org/TR/WGSL/#const-decls) for `fill_color_id`
```wgsl
const fill_color_id : i32 = -1;

struct Layout {
    box_transtionation: mat4x4<f32>,
    texture_id: i32,
    fill_color: vec4<f32> // used only when texture_id == fill_color_id
}
```